### PR TITLE
Refactor InstanceCreate

### PR DESCRIPTION
**Pull request by hidekazuna**
**What this PR does / why we need it**:

This PR is only refactoring, deletes network and subnet check logic from InstanceCreate.

Since openstackmachine controller checks whether infrastructure ready or not first, network and subnet are created before InstanceCreate.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hidekazuna** [comment]
recheck

**Comment by hidekazuna** [comment]
recheck

**Comment by hidekazuna** [comment]
recheck

File: pkg/cloud/services/compute/instance.go [hunk 1]
```diff
@@ -102,30 +100,10 @@ func (s *Service) InstanceCreate(clusterName string, machine *clusterv1.Machine,
 		}
 	}
 
-	// Get all network UUIDs
-	var nets []ServerNetwork
-	if len(openStackMachine.Spec.Networks) > 0 {
-		var err error
```
Comment (jichenjc) [inline pkg/cloud/services/compute/instance.go:108]: I assume this change will ignore the OpenstackMachine Spec networks and only honor the cluster status network
then what's the reason to keep OpenstackMachine Spec networks?

Comment (hidekazuna) [reply pkg/cloud/services/compute/instance.go:108]: Aha, I understand openStackMachine.Spec.Networks is used the instance is connected to multi networks.  I wonder if there is really useful though.

Comment (jichenjc) [reply pkg/cloud/services/compute/instance.go:108]: @hidekazuna I think multiple network is valid ...

**Pull request closed without merging**